clusterloader2/pkg/prometheus/experimental.go

```diff
@@ -75,6 +80,34 @@ func (pc *PrometheusController) trySnapshotPrometheusDisk() (bool, error) {
 		klog.Info(string(s))
 		return true, nil
 	}
+	pc.diskMetadata.name = pdName
+	pc.diskMetadata.zone = zone
+	return true, nil
+}
+
+func (pc *PrometheusController) snapshotPrometheusDiskIfEnabled() error {
+	if !*shouldSnapshotPrometheusDisk {
+		return nil
+	}
+	if pc.provider != "gce" && pc.provider != "gke" && pc.provider != "kubemark" {
+		return fmt.Errorf(
+			"snapshotting Prometheus' disk only available for GCP providers (gce, gke, kubemark), provider is: %s", pc.provider)
+	}
+	// Update cache of Prometheus disk metadata
+	err := wait.Poll(
+		10*time.Second,
+		2*time.Minute,
+		pc.tryRetrievePrometheusDiskMetadata)
```

Comment:
See the comment below first.

Alternatively (to not have to wait.Polls) you can call tryRetrievePrometheusDiskMetadata from trySnaphostPrometheusDisk and only retry the latter.

Reply:
done with separate Poll()